### ISSUE TITLE
The failed_jobs insert error

### ISSUE THREAD

**Issue by tbryan24** [mongodb/laravel-mongodb#2379]
- Laravel-mongodb Version: #.#.#
- PHP Version: #.#.#
- Database Driver & Version:

### Steps to reproduce
1.add a job
2.throw exception in job
3.

### Expected behaviour
There has error if insert more than one record into failed_jobs table 
error message : duplicate key error collection: life_moments.failed_jobs index: uuid_1 dup key: { uuid: null }

### Actual behaviour
Rewrite log method from MongoFailedJobProvider.php
add 
public function log($connection, $queue, $payload, $exception)
    {
        $failed_at = Carbon::now()->getTimestamp();

        $uuid = json_decode($payload, true)['uuid'];//add this ,because uuid is unique

        $exception = (string) $exception;

        $this->getTable()->insert(compact('uuid','connection', 'queue', 'payload', 'failed_at', 'exception'));
    }

<details><summary><b>Logs</b>:</summary>
Insert log.txt here (if necessary)
</details>

**Comment by playerarm123** [comment]
I have same problem

**Comment by BehroozBvk** [comment]
Hi, I also had this problem and **solved** it.

Before telling the solution, I must say that if the configuration considered for `failed.driver` in the `queue.php` file is `database-uuids` by default, the value of the `failed_at` field in Mongo is an empty object and no date is stored in it.

But if we change the driver to **mongodb**, you will see that the date is stored as a **string** of the `timestamp` value in the `failed_at` field. And this was not what I expected and I wanted the date to be stored as an **ISODate**.

However, to save the date in the `failed_at` field as **ISODate**, you can take the following steps.

Step 1:
-------
Run the following command 
```bash
php artisan make:provider CustomMongodbQueueServiceProvider
```
Then rewrite the `CustomMongodbQueueServiceProvider.php` file as below:

```php
<?php

namespace App\Providers;

use Jenssegers\Mongodb\MongodbQueueServiceProvider;

class CustomMongodbQueueServiceProvider extends MongodbQueueServiceProvider
{
    /**
     * Create a new MongoDB failed job provider.
     */
    protected function mongoFailedJobProvider(array $config): CustomMongoFailedJobProvider
    {
        return new CustomMongoFailedJobProvider($this->app['db'], $config['database'], $config['table']);
    }
}
```

Step 2:
-------
Run the following command :
```bash
php artisan make:provider CustomMongoFailedJobProvider
```
Then rewrite the `CustomMongoFailedJobProvider.php` file as below:

```php
<?php

namespace App\Providers;

use Jenssegers\Mongodb\Queue\Failed\MongoFailedJobProvider;

class CustomMongoFailedJobProvider extends MongoFailedJobProvider
{
    public function log($connection, $queue, $payload, $exception): void
    {
        $failed_at = new \MongoDB\BSON\UTCDateTime(now());
        $uuid = json_decode($payload, true)['uuid'];
        $exception = (string) $exception;
        $this->getTable()->insert(compact('connection', 'uuid', 'queue', 'payload', 'failed_at', 'exception'));
    }
}
```

I think it is generally clear what we did, a simple override of two service providers related to `MongoFailedJobProvider`, `MongodbQueueServiceProvider` through which we can rewrite the `log` method according to our needs.

Step 3:
-------
Now we have to register our service provider in the `config/app.php` file :
```php
'providers' => [
    // Other Service Providers ...
    App\Providers\CustomMongodbQueueServiceProvider::class,
],
```


Step 4:
-------
Then run the following command :
```bash
composer dump-autoload
```

Well, our work is finished, now you can test the result. I hope this solution helped you.

![photo_2023-09-06_17-01-47](https://github.com/mongodb/laravel-mongodb/assets/17563730/3bd3961a-f45b-4c6c-ae69-462533463296)

**Comment by GromNaN** [comment]
https://jira.mongodb.org/browse/PHPORM-87

**Comment by GromNaN** [comment]
Fixed by #2838
